src/main/java/org/apache/sling/servlets/post/impl/helper/SortedByRankingList.java

```diff
@@ -0,0 +1,106 @@
+/*
```

Comment:
This looks like it could be replaced with an implementation of [SortedMap] rather than a new, custom class. A benefit of using a SortedMap is that you can use either a concurrency safe version or not, depending on the use case.

[SortedMap]: https://docs.oracle.com/javase/8/docs/api/java/util/SortedMap.html

Reply:
Hi Paul, great point, I kept the same insertion code used before just moving it to the new Class, so it could reused.

I can reimplement it using a SortedMap as proposed. Although I am not sure we can guarantee if no ranking is provided, the last inserted element will be the chosen one.

Reply:
I would check what the OSGi core bundles do - I think that the 'sort by ranking' problem occurs there as well. @cziegeler / @karlpauls / @bosschaert  - any suggestions?

Reply:
Please rather use DS field injection which orders automatically by ranking

> For a reference with multiple cardinality, a field must be a collection of one of the following types:
Collection
List
A subtype of Collection - This type can only be used for dynamic references using the update reference field option. The component instance must initialize the field to a collection object in its constructor.
(https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-field.injection)

Reply:
Thank you for the feedback! I will rework on this PR and reimplement the binding using the references given. I believe it is straightforward to implement it for NodeNameGenerator, SlingPostProcessor and PostResponseCreator lists.

Although, I need to rethink how to bind/unbind the postOperation based on OperationName, as in the implementation we have a Map(operationName, List of PostOperation services). Having one large list is not optimal and the map was useful.

If any of you have an idea how I can solve the map problem, I accept suggestions 😄